src/blog/2022/12/flowforge-gcp-https-set-up.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Configure FlowForge in Docker to secure all traffic
+subtitle: Use Let's Encrypt and Acme Companion to quickly set up FlowForge to encrypt all traffic
+description: Use Let's Encrypt and Acme Companion to quickly set up FlowForge to encrypt all traffic
+date: 2022-12-09
+authors: ["rob-marcer"]
+---
+
+
+Following on from our [previous article](https://flowforge.com/blog/2022/10/ff-docker-gcp/) in which we covered how to run FlowForge in Docker on Google’s Cloud Platform, today we are going to look at how to secure HTTP traffic to your FlowForge server.
+
+<!--more-->
+
+### Introduction
+
+When we wrote the first part of this series FlowForge didn't have an easy path to secure HTTP traffic. Happily, two versions of FlowForge later and at least partially inspired by these blogs, we have added the configuration you need in Docker to use HTTPS with minimal work.
+
+That addition makes our job of explaining this setup a lot easier, credit to our developers for seeing the value of having an easy implementation of HTTPS in FlowForge as part of our [1.0 build](https://flowforge.com/blog/2022/10/flowforge-1-released/).
+
+To achieve secure HTTPS traffic we are employing a great service called [Let's Encrypt](https://letsencrypt.org/). In their own words, "Let’s Encrypt is a free, automated, and open certificate authority (CA), run for the public’s benefit". In practice Let's Encrypt will allow us to easily secure HTTPS traffic.
+
+We are also utilising a Docker image called [acme-companion](https://github.com/nginx-proxy/acme-companion) which makes the configuration of Let's Encrypt a breeze. To quote the project's own Github page "acme-companion is a lightweight companion container for nginx-proxy. It handles the automated creation, renewal and use of SSL certificates for proxied Docker containers through the ACME protocol". It's a great project and credit to the team over there for making it a lot easier to secure the internet.
+
+Now we've covered our goals and the tools we are going to use let's configure our existing GCP VM to secure all web traffic.
+
+### Update Docker Compose
+
+The first step is to edit our Docker compose file. We're using Nano again to edit files so we will run this command:
+
+```
+sudo nano /opt/flowforge/docker-compose-1.1.1/docker-compose.yml
```

Comment:
Is a pre-requisite that the customer upgraded their docker to 1.1? If so, let's say so and link to the upgrade docs

Reply:
No, it was added in 1.0 which I have said at the top but do you think it would be better to state it as a clear pre-requisite?

Reply:
I've added 1.0 as a clear prerequisites